plugins/api/src/switch_user.cpp

```diff
@@ -0,0 +1,198 @@
+#include "irods/plugins/api/api_plugin_number.h"
+#include "irods/plugins/api/switch_user_types.h"
+#include "irods/rodsDef.h"
+#include "irods/rcConnect.h"
+#include "irods/rodsPackInstruct.h"
+#include "irods/rcMisc.h"
+#include "irods/client_api_allowlist.hpp"
+
+#include "irods/apiHandler.hpp"
+
+#include <functional>
+
+#ifdef RODS_SERVER
+
+//
+// Server-side Implementation
+//
+
+#  include "irods/irods_rs_comm_query.hpp"
+#  include "irods/rodsErrorTable.h"
+#  include "irods/irods_logger.hpp"
+#  include "irods/server_utilities.hpp"
+
+#  define IRODS_USER_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#  include "irods/user_administration.hpp"
+
+#  include <cstring>
+#  include <string>
+#  include <string_view>
```

Comment:
Not a fan of the indentation here because this preprocessor block contains more than just preprocessor directives. If these directives were closer to the `#ifdef`, I'd be happier.

I wonder what others think about this?

Reply:
Honestly I think I've decided that I don't like the way `clang-format` does preprocessor indentation at all. I wouldn't be opposed to switching it back off.

Reply:
I haven't reached a decision about the formatting of PP directives just yet. Will think about it a bit more as this PR nears completion.

Reply:
Created issue: #6644